.github/workflows/reusable-tests.yml

```
@@ -0,0 +1,62 @@
+on:
+  workflow_call:
+    inputs:
+      profile:
+        required: true
+        type: string
+      # Parameters deciding which tests to run
+      test-tari:
+        required: true
+        type: boolean
+      test-wasm:
+        # flag to run any wasm tests. Includes key manager
+        required: true
+        type: boolean
+
+
+jobs:
+  tests:
+    name: test
+    runs-on: ubuntu-18.04
```

Or is it better to replace all of these?
```suggestion
    runs-on: [ self-hosted, ubuntu18.04-high-cpu ]
```